src/test/test_chunk_processors.py

```diff
@@ -749,3 +749,164 @@ def test_split_text_into_sentences_complex_cases(input_chunks, expected_text):
+            id="7",
+        ),
+        Chunk(
+            text="Page 3",
+            chunk_type=BlockType.PAGE_HEADER,
+            bounding_boxes=None,
+            pages=[3],
+            id="8",
+        ),
+        Chunk(
+            text="and finishes on page 3.",
+            chunk_type=BlockType.TEXT,
+            bounding_boxes=None,
+            pages=[3],
+            id="9",
+        ),
+    ]
+
+    result = processor(chunks)
+
+    # 3 sentences + 4 header/footer chunks
+    assert len(result) == 7
+
+    # TODO: the current logic doesn't preserve sentence positions relative to ignored
+    # chunks, when the sentences are across multiple chunks.
+    # This means that the final order is
+    # - Page 1 footer
+    # - Page 2 header
+    # - Page 1 & 2 text
+    # - Page 2 text
+    # - Page 2 footer
+    # - Page 3 header
+    # - Page 2 & 3 text
+
+    # Check that the ignored chunks are preserved
+    # assert result[1].chunk_type == BlockType.PAGE_FOOTER
+    # assert result[2].chunk_type == BlockType.PAGE_HEADER
+    # assert result[5].chunk_type == BlockType.PAGE_FOOTER
+    # assert result[6].chunk_type == BlockType.PAGE_HEADER
+
+    assert (
+        result[2].text
+        == "This is the beginning of a sentence that continues across page boundaries."
+    )
+    assert result[2].pages == [1, 2]
+
+    assert result[3].text == "This is a complete sentence on page 2."
+    assert result[3].pages == [2]
+
+    assert result[6].text == "This sentence starts on page 2 and finishes on page 3."
+    assert result[6].pages == [2, 3]
+
+
+@pytest.mark.skip(reason="This is an unknown issue with the current sentence splitter.")
+def test_split_text_into_sentences_with_footnote_between_sentence_parts():
+    """Test that sentences can span across chunks separated by a footnote."""
+    processor = SplitTextIntoSentences()
+    chunks = [
+        Chunk(
+            text="They want a united country, based on democratic principles,",
+            chunk_type=BlockType.TEXT,
+            bounding_boxes=None,
+            pages=[11],
+            id="35",
+        ),
+        Chunk(
+            text='A Steering Committee, led by the Minister of Development and Economic Planning, supervised the consultations and gave technical direction. The work was supported by UNDP\'s Sierra Leone Office and "African Futures", a UNDP regional project based in Abidjan.',
+            chunk_type=BlockType.FOOT_NOTE,
+            bounding_boxes=None,
+            pages=[11],
+            id="36",
+        ),
+        Chunk(
+            text="rule of law, and justice for all, whose citizens participate actively in national and local management; a dynamic, open, enlightened, integrated society. People called for a new type of leadership - responsible, responsive, effective, and accountable.",
+            chunk_type=BlockType.TEXT,
+            bounding_boxes=None,
+            pages=[12],
+            id="37",
+        ),
+    ]
```

Comment:
This is the failure case, marked with a skipped test for now. I think it's lower priority than getting the fix in but I wanted to record the issue.